### PR TITLE
Ask before closing the wizard

### PR DESCRIPTION
**Pull request by TheOneRing**
Fixes: #10232

### CODE REVIEW

File: src/gui/newwizard/setupwizardwindow.cpp [hunk 1]
```diff
@@ -149,6 +151,18 @@ void SetupWizardWindow::slotStartTransition()
     slotHideErrorMessageWidget();
 }
 
+void SetupWizardWindow::reject()
+{
+    auto messageBox = new QMessageBox(QMessageBox::Warning, tr("Cancel Setup"), tr("Do you really want to cancel the account setup?"), QMessageBox::Yes | QMessageBox::No, ocApp()->gui()->settingsDialog());
```
Comment (fmoc) [inline src/gui/newwizard/setupwizardwindow.cpp:156]: This message doesn't make a lot of sense to me terminology wise. We consider this the "setup wizard".

File: src/gui/newwizard/setupwizardwindow.cpp [hunk 2]
```diff
@@ -35,6 +36,7 @@ SetupWizardWindow::SetupWizardWindow(SettingsDialog *parent)
     , _ui(new ::Ui::SetupWizardWindow)
 {
     Utility::setModal(this);
+    setWindowFlag(Qt::WindowCloseButtonHint, false);
```
Comment (fmoc) [inline src/gui/newwizard/setupwizardwindow.cpp:39]: Why do you need this? Closing the dialog via the regular exit button added by the window manager also triggers `reject`.

Apparently, on macOS one needs to combine this with `WindowSystemMenuHint`.

**Review by fmoc — CHANGES_REQUESTED** [review]
I think we need to complement the "you're all set" message with a "press finish to actually finish" kind of message.

File: src/gui/newwizard/setupwizardwindow.cpp (hunk 1 above)
Comment (TheOneRing) [reply src/gui/newwizard/setupwizardwindow.cpp:156]: Yes but the wizard does set up an account?

File: src/gui/newwizard/setupwizardwindow.cpp (hunk 2 above)
Comment (TheOneRing) [reply src/gui/newwizard/setupwizardwindow.cpp:39]: This disables the X on Windows, I think we give better guidance if we force the user to use the close button?

**Comment by TheOneRing** [comment]
> I think we need to complement the "you're all set" message with a "press finish to actually finish" kind of message.

I expect 99% of the users will figure it out and for the rest we have the cancel dialog.

File: src/gui/newwizard/setupwizardwindow.cpp (hunk 2 above)
Comment (fmoc) [reply src/gui/newwizard/setupwizardwindow.cpp:39]: You mean the cancel button? I'm not sure... I like to use either option. But I'm a dev.

> Apparently, on macOS one needs to combine this with WindowSystemMenuHint.

My bad, I guess with these sheets one doesn't have to care about this anyway.

**Pull request merged**